### PR TITLE
Check prereleases testing with macOS + Python 3.12 + PyQt6

### PR DESCRIPTION
**Pull request by dalthviz**
# References and relevant issues

Follow up for #6826 (macOS + Python 3.12 + PyQt6 case)

# Description

Explore enabling macOS with Python 3.12 and PyQt6 case for prereleases testing

### CODE REVIEW

**Comment by Czaki** [comment]
@dalthviz Maybe you should disable test workflow by editing `.github/workflow/test_pull_request.yml` to speedup CI and reduce obsolete runs?

**Comment by dalthviz** [comment]
Thanks @Czaki for the suggestion! But I'm not totally sure to understand the idea 😅 

As a quick summary of the PR checks state, I think that all the prerelease checks are passing now :) However, checking [the last error shown over the checks](https://github.com/napari/napari/actions/runs/8803786746/job/24163311438?pr=6858#step:12:190), seems like it comes from a `pyqt5-qt5==5.15.2` constraint (which maybe comes from the constraint files?). Since now `macos-latest` ends up being macOS 14 arm64 and there are only macOS intel wheels for `pyqt5-qt5==5.15.2` pip ends up showing a `ResolutionImpossible` error 🤔

Could it be that disabling the pull request test workflow could help with that? The only thing that occurs to me is to change from `macos-latest` image to some other image (`macos-12`, `macos-13`, `macos-latest-large` or [some other](https://github.com/actions/runner-images?tab=readme-ov-file#available-images)) 🤔 

Let me know what do you think!

**Comment by Czaki** [comment]
> As a quick summary of the PR checks state, I think that all the prerelease checks are passing now :) However, checking [the last error shown over the checks](https://github.com/napari/napari/actions/runs/8803786746/job/24163311438?pr=6858#step:12:190), seems like it comes from a `pyqt5-qt5==5.15.2` constraint (which maybe comes from the constraint files?). Since now `macos-latest` ends up being macOS 14 arm64 and there are only macOS intel wheels for `pyqt5-qt5==5.15.2` pip ends up showing a `ResolutionImpossible` error 🤔

It is surprising for me that pyqt5 has macos arm wheels but pyqt5-qt which is dependency of pyqt5 does not have such wheels..

> Could it be that disabling the pull request test workflow could help with that? The only thing that occurs to me is to change from `macos-latest` image to some other image (`macos-12`, `macos-13`, `macos-latest-large` or [some other](https://github.com/actions/runner-images?tab=readme-ov-file#available-images)) 🤔

no. It was for test sped up as repo has low level limits on macos runners, But as you resolved the problem it is not needed.

**Comment by psobolewskiPhD** [comment]
> It is surprising for me that pyqt5 has macos arm wheels but pyqt5-qt which is dependency of pyqt5 does not have such wheels..
> 
> >

What's really odd is that newer releases *only* have macOS wheels:
https://pypi.org/project/PyQt5-Qt5/5.15.13/#files

File: .github/workflows/test_pull_requests.yml [hunk 1]
```diff
@@ -125,7 +125,7 @@ jobs:
             backend: pyqt6
             coverage: no_cov
           - python: "3.12"
-            platform: macos-latest
+            platform: macos-12
```
Comment (Czaki) [inline .github/workflows/test_pull_requests.yml]:
```suggestion
            platform: macos-13
```

13 should be also x86-64? Or you tested it?

**Comment by Czaki** [comment]
> What's really odd is that newer releases _only_ have macOS wheels:

it is my bad. There were two packages. pyqt5-qt and pyqt5-qt5 and pyqt5-qt is deprecated. 

But pyqt5-qt5 release 5.15.2 is for all systems, except macos-arm. And 5.15.11, 5.15.12, and 5.15.13 are macos only, because of this, our pinning dependencies do not pick them.

I have send mail to PyQt5 maintainer with ask about Linux and Widows wheels for PyQt5-Qt5==5.15.13

**Review by Czaki — APPROVED** [review]
looks ok. Thanks for your big effort.

File: .github/workflows/test_pull_requests.yml (hunk 1 above)
Comment (psobolewskiPhD) [reply .github/workflows/test_pull_requests.yml]: macOS-13 has x64 runner, but the question is what is the default?
macOS-14 has clearly switched to arm64 default, but also has x64.

Comment (Czaki) [reply .github/workflows/test_pull_requests.yml]: https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories

based on this information, the macos-13 means x86 runner and macos-14/macos-latest means arm runner.

Comment (dalthviz) [reply .github/workflows/test_pull_requests.yml]: Checking the GitHub runner-images repo (https://github.com/actions/runner-images?tab=readme-ov-file#available-images) I think `macos-13` means x86 indeed :+1:

**Pull request merged**